### PR TITLE
add internal SSL

### PR DESCRIPTION
**Pull request by kellyrowland**

### CODE REVIEW

**Comment by rcthomas** [comment]
I've tested on the compose test problem in our deploy repo.  Whew.  The only request I have is that the poll() method revert to the PID check.  We can worry about moving the cert setup outside start() later, when we drop the pool of hosts feature (or otherwise change it to be determined before create/move_certs happens).

File: sshspawner/sshspawner.py [hunk 1]
```diff
@@ -141,15 +175,49 @@ def clear_state(self):
             self.clear_state()
             return 0
 
+        # set up ssl context and check
```
Comment (rcthomas) [inline sshspawner/sshspawner.py]: Please revert to PID check thanks!!!

**Review by rcthomas — CHANGES_REQUESTED** [review]

File: sshspawner/sshspawner.py (hunk 1 above)
Comment (kellyrowland) [reply sshspawner/sshspawner.py]: done ✔️

File: sshspawner/sshspawner.py [hunk 2]
```diff
@@ -99,16 +107,41 @@ def clear_state(self):
...
         self.remote_ip, port = await self.remote_random_port()
         if self.remote_ip is None or port is None or port == 0:
             return False
+        self.remote_port = str(port)
         cmd = []
 
         cmd.extend(self.cmd)
         cmd.extend(self.get_args())
 
+        with TemporaryDirectory() as td:
```
Comment (rcthomas) [inline sshspawner/sshspawner.py]: One other thing; this should be blocked off if internal ssl is not set.  I think that's a temporary fix here.  Once this is merged we can move it around and get rid of the multi-host stuff to make it work with move certs hook.  Should have caught this yesterday.

**Review by rcthomas — CHANGES_REQUESTED** [review]

File: sshspawner/sshspawner.py (hunk 2 above)
Comment (kellyrowland) [reply sshspawner/sshspawner.py]: added conditional and tested locally, should be good to go. thanks for the thorough review!

**Review by rcthomas — APPROVED** [review]

**Comment by kellyrowland** [comment]
🎉

**Pull request merged**